sahi/utils/coco.py

```diff
@@ -127,15 +127,15 @@ def from_coco_annotation_dict(cls, annotation_dict: Dict, category_name: Optiona
             annotation_dict: dict
                 COCO formatted annotation dict (with fields "bbox", "segmentation", "category_id")
         """
-        if annotation_dict["segmentation"] and not isinstance(annotation_dict["segmentation"], list):
+        if annotation_dict.__contains__("segmentation") and not isinstance(annotation_dict["segmentation"], list):
             has_rle_segmentation = True
             logger.warning(
                 f"Segmentation annotation for id {annotation_dict['id']} is skipped since RLE segmentation format is not supported."
             )
         else:
             has_rle_segmentation = False
 
-        if annotation_dict["segmentation"] and not has_rle_segmentation:
+        if annotation_dict.__contains__("segmentation") and not has_rle_segmentation:
```

Comment:
similarly,

here we have to check if `segmentation` field is not empty (object detection datasets in COCO format have `segmentation: []` in the json file)

Reply:
I thought that only there is the check necessary